### ISSUE TITLE
how to change port in docker-compose deployment

### ISSUE THREAD

**Issue by ZTurboX** [heartexlabs/label-studio#2646]
how to change port use docker-compose deploy

**Comment by farioas** [comment]
It well declared in docker-compose documentation:
https://docs.docker.com/compose/compose-file/#ports

```
    ports:
      - "8080:80"
      - "8081:443"
```

On the left side exposed ports to host, on the right side ports within docker. You shouldn't modify ports on the right side.

**Comment by WillieMaddox** [comment]
@farioas According to the docker-compose documentation, the order is `HOST:CONTAINER`.

**Comment by farioas** [comment]
@WillieMaddox good catch, I updated my comment

**Comment by WillieMaddox** [comment]
@farioas Thanks for the quick reply.  As a follow up, In my `docker-compose.yml` I changed the `nginx` http port from `8080:8086` to `9876:8086`.  What then should the `app` expose port be set to? The same a the `nginx` host port, `9876`? or leave it as the default `8080`? As far as I can tell, it makes no difference: As long as it hit localhost:9876, everything seems to be working correctly.  What am I missing?

**Comment by farioas** [comment]
No changes in app is required. Since nginx is a front-facing module, so all traffic comes through it.

**Comment by TiansuYu** [comment]
What if a port is ready exposed in the Dockerfile
```
EXPOSE 1234
```
How could i modify that binding e.g. in docker compose to 
```
ports:
   - "4321:1234"
```
I tried this, but it does not work. After `docker compose up`, the port is still binds to `localhost:1234`.

**Comment by mcint** [comment]
> How could i modify that binding e.g. in docker compose to
> ```
> ports:
>    - "4321:1234"
> ```
> 
> I tried this, but it does not work. After `docker compose up`, the port is still binds to `localhost:1234`.

I've encountered a similar issue, in another project. Conceptually it seems "docker compose" management layer treats port mapping or exported ports as a built-in part of the configuration. I ran a `docker compose restart` and did not observe changes, but when I ran `docker compose up` again, as you mentioned, and as directed by [docker compose docs: update containers on the network](https://docs.docker.com/compose/how-tos/networking/#update-containers-on-the-network), I observed the container with forwarded ports (and two others) recreated. The new port mappings are present after new `up`.
